types/firestore.d.ts

```diff
@@ -1001,40 +1066,28 @@ declare namespace FirebaseFirestore {
      */
     isEqual(other: CollectionReference): boolean;
 
-    /**
-     * Creates and returns a new CollectionReference instance that applies a
-     * field mask and only returns the specified subset of fields for calls to
-     * `get()`, `stream()`, `getAll()` and `streamAll()`. You can specify a list
-     * of field paths to return, or use an empty list to only return the
-     * references of matching documents.
-     *
-     * This function returns a new (immutable) instance of the
-     * CollectionReference (rather than modify the existing instance) to impose
-     * the field mask.
-     *
-     * @param field The field paths to return.
-     * @return The created Query.
-     */
-    select(...field: (string | FieldPath)[]): CollectionReference;
-
     /**
      * Retrieves all documents in this CollectionReference.  Documents with
      * subcollections that contain no other data are returned as
-     * `DocumentSnapshots` whose `exists` property returns false.
+     * `DocumentSnapshots` whose `exists` property is set to false.
      *
+     * @param options An object to configure the behavior of this `getAll` call
+     * (i.e. the set of fields to return).
      * @return A Promise that will be resolved with the contents of of this
      * collection.
      */
-    getAll(): Promise<DocumentSnapshot>;
+    getAll(options?:GetOptions): Promise<DocumentSnapshot>;
```

Comment:
I'm not sure I'm a fan of re-using the name getAll(). This method seems substantively different to me since the arguments are different (it enumerates all documents rather than retrieving a set of specified documents).